internal/cephfs/mounter/kernel.go

```diff
@@ -27,13 +29,47 @@ import (
 const (
 	volumeMounterKernel = "kernel"
 	netDev              = "_netdev"
+	kernelModule        = "ceph"
 )
 
-type KernelMounter struct{}
+// testErrorf can be set by unit test for enhanced error reporting.
+var testErrorf = func(fmt string, args ...any) { /* do nothing */ }
 
-func mountKernel(ctx context.Context, mountPoint string, cr *util.Credentials, volOptions *store.VolumeOptions) error {
-	if err := execCommandErr(ctx, "modprobe", "ceph"); err != nil {
-		return err
+type KernelMounter interface {
+	Mount(
+		ctx context.Context,
+		mountPoint string,
+		cr *util.Credentials,
+		volOptions *store.VolumeOptions,
+	) error
+
+	Name() string
+}
+
+type kernelMounter struct {
+	// needsModprobe indicates that the ceph kernel module is not loaded in
+	// the kernel yet (or compiled into it)
+	needsModprobe bool
+}
+
+func NewKernelMounter() KernelMounter {
+	return &kernelMounter{
+		needsModprobe: !filesystemSupported(kernelModule),
+	}
+}
+
+func (m *kernelMounter) mountKernel(
+	ctx context.Context,
+	mountPoint string,
+	cr *util.Credentials,
+	volOptions *store.VolumeOptions,
+) error {
+	if m.needsModprobe {
+		if err := execCommandErr(ctx, "modprobe", kernelModule); err != nil {
+			return err
+		}
+
+		m.needsModprobe = false
```

Comment:
i think this needs to be set to true? also this is not of much use with the current code because mounter.New() is called for every NodeStage RPC call. we need to move this to Driver initialization time as it needs to run only one time.

Reply:
once the module is loaded, it will be listed in `/proc/filesystems`, so eventually `modprobe` is only run once